tests/testImageDecorrelation.py

```diff
@@ -0,0 +1,253 @@
+
+    im1_psf = singleGaussian2d(x0im, y0im, 0, 0, psf1[0], psf1[1], theta=theta1)
+    im2_psf = singleGaussian2d(x0im, y0im, offset[0], offset[1], psf2[0], psf2[1], theta=theta2)
+    return im1, im2, im1_psf, im2_psf
+
+
+class DiffimCorrectionTest(lsst.utils.tests.TestCase):
+    """!A test case for the diffim image decorrelation algorithm.
+    """
+
+    def setUp(self):
+        """!Generate a fake aligned template and science image and analyse the noise in the
+        resulting diffim. First, use a non-spatially-varying psf.
+        """
+
+        self.psf1 = 2.2  # sigma of psf of template image
+        self.psf2 = 3.3  # sigma of psf of science image
+        self.sig1 = 0.2  # std.dev of noise in template image
+        self.sig2 = 0.2  # std.dev of noise in science image
+
+        self.im1, self.im2, self.im1_psf, self.im2_psf \
+            = makeFakeImages(sig1=self.sig1, sig2=self.sig2, psf1=self.psf1, psf2=self.psf2,
+                             n_sources=50, verbose=True)
+        bbox = afwGeom.Box2I(afwGeom.Point2I(0, 0), afwGeom.Point2I(self.im1.shape[0]-1, self.im1.shape[1]-1))
+
+        # All this code to convert the template image array/psf array into an exposure.
+        self.im1ex = afwImage.ExposureF(bbox)
+        self.im1ex.getMaskedImage().getImage().getArray()[:, :] = self.im1
+        self.im1ex.getMaskedImage().getVariance().getArray()[:, :] = self.sig1**2.
+        psfBox = afwGeom.Box2I(afwGeom.Point2I(-20, -20), afwGeom.Point2I(20, 20))
+        psf = afwImage.ImageD(psfBox)
+        psfBox.shift(afwGeom.Extent2I(256, 256))
+        im1_psf_sub = self.im1_psf[psfBox.getMinX():psfBox.getMaxX()+1, psfBox.getMinY():psfBox.getMaxY()+1]
+        psf.getArray()[:, :] = im1_psf_sub
+        psfK = afwMath.FixedKernel(psf)
+        psfNew = measAlg.KernelPsf(psfK)
+        self.im1ex.setPsf(psfNew)
+
+        # Now for the 2nd (science) image
+        self.im2ex = afwImage.ExposureF(bbox)
+        self.im2ex.getMaskedImage().getImage().getArray()[:, :] = self.im2
+        self.im2ex.getMaskedImage().getVariance().getArray()[:, :] = self.sig2**2.
+        psf = afwImage.ImageD(psfBox)
+        im2_psf_sub = self.im2_psf[psfBox.getMinX():psfBox.getMaxX()+1, psfBox.getMinY():psfBox.getMaxY()+1]
+        psf.getArray()[:, :] = im2_psf_sub
+        psfK = afwMath.FixedKernel(psf)
+        psfNew = measAlg.KernelPsf(psfK)
+        self.im2ex.setPsf(psfNew)
+
+    def tearDown(self):
+        del self.im1
+        del self.im1_psf
+        del self.im2
+        del self.im2_psf
+        del self.im1ex
+        del self.im2ex
+
+    def testDiffimCorrection(self):
+        """! Check that the variance of the corrected diffim matches the theoretical value
+        (to within a 1% tolerance).
+        """
+
+        # Create the matching kernel. We use gaussians, so it's just:
+        sig_match = np.sqrt((self.psf2**2. - self.psf1**2.))
+        matchingKernel = measAlg.SingleGaussianPsf(25, 25, sig_match)
```

Comment:
Sorry, this was a little confusing. The matching kernel in this case is Gaussian because the (input) PSFs of the two synthesized images are Gaussian. The PSF1 and PSF2 sigma's are already stored in setUp as psf1 and psf2. But now I will just compute them from the exposures' PSF's shapes as you suggested.